### PR TITLE
ClusterDiscovery supports setting username and password

### LINKED ISSUE #58063: How to set password for distributed table in cluster discovery mode
I have a ck cluster with `users.xml` to specified password for users. But when I try to query a distributed table on it, I got exception:
```
10.132.174.22 :) select * from db1.table1_dist;                                                                                                                                                                      
                                                                                                                                                                                                                     
SELECT *                                                                                                                                                                                                             
FROM db1.table1_dist                                                                                                                                                                                                 
                                                                                                                                                                                                                     
Query id: 3a0269d4-6048-463c-b442-8e9861ab505d                                                                                                                                                                       
                                                                                                                                                                                                                     
Connecting to localhost:9000 as user default.                                                                                                                                                                        
Exception on client:                                                                                                                                                                                                 
Code: 516. DB::Exception: Received from localhost:9000. DB::Exception: default: Authentication failed: password is incorrect, or there is no user with such name.                                                    
                                                                                                                                                                                                                     
If you have installed ClickHouse and forgot password you can reset it in the configuration file.                                                                                                                     
The password for default user is typically located at /etc/clickhouse-server/users.d/default-password.xml                                                                                                            
and deleting this file will reset the password.                                                                                                                                                                      
See also /etc/clickhouse-server/users.xml on the server where ClickHouse is installed.                                                                                                                               
                                                                                                                                                                                                                     
. (AUTHENTICATION_FAILED)     
```

What I know is to config user and password in `remote_servers.replica`, just like #5823 discuss. But I am deploy mode in cluster discovery mode like [this](https://clickhouse.com/docs/en/operations/cluster-discovery#use-cases-and-limitations) which seems no properties about username/password.

Thx.

### PR DESCRIPTION
**Pull request by vdimir**
### Changelog category (leave one):

- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

* ClusterDiscovery supports setting username and password, close https://github.com/ClickHouse/ClickHouse/issues/58063

### Documentation entry for user-facing changes

- [x] Documentation is written (mandatory for new features)

### CODE REVIEW

**Comment by robot-clickhouse-ci-2** [comment]
*This is an automated comment for commit e29c92569ec7b31d54fa577bb0254439a6cdd2ff with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/58123/e29c92569ec7b31d54fa577bb0254439a6cdd2ff/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/58123/e29c92569ec7b31d54fa577bb0254439a6cdd2ff/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

File: tests/integration/test_cluster_discovery/test_password.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+
+def test_connect_with_password(start_cluster):
+    check_on_cluster(
+        [nodes["node0"], nodes["node1"]],
+        len(nodes),
+        cluster_name="test_auto_cluster_with_pwd",
+        what="count()",
+        msg="Wrong nodes count in cluster",
+        query_params={"password": "passwordAbc"},
```
Comment (antonio2368) [inline tests/integration/test_cluster_discovery/test_password.py:43]: maybe add a test with invalid password?

Comment (vdimir) [reply tests/integration/test_cluster_discovery/test_password.py:43]: Added cluster configuration with wrong password. (I still have to pass password=passwordAbc, but checking that password from cluster configuration is used for inter-server communication)

**Comment by vdimir** [comment]
Also added support of `<secret>` instead of `<password>`

File: src/Interpreters/ClusterDiscovery.cpp [hunk 2]
```diff
@@ -124,21 +125,33 @@ ClusterDiscovery::ClusterDiscovery(
...
             continue;
 
+        String zk_root = config.getString(cluster_config_prefix + ".path");
+        if (zk_root.empty())
+            throw Exception(ErrorCodes::NO_ELEMENTS_IN_CONFIG, "ZooKeeper path for cluster '{}' is empty", key);
+
+        const auto & password = config.getString(cluster_config_prefix + ".password", "");
+        const auto & cluster_secret = config.getString(cluster_config_prefix + ".secret", "");
+        if (!password.empty() && !cluster_secret.empty())
+            throw Exception(ErrorCodes::NO_ELEMENTS_IN_CONFIG, "Mutually exclusive options 'password' and 'secret' are specified for cluster '{}'", key);
```
Comment (antonio2368) [inline src/Interpreters/ClusterDiscovery.cpp:139]: Let's make it more human friendly
```suggestion
            throw Exception(ErrorCodes::NO_ELEMENTS_IN_CONFIG, "Both 'password' and 'secret' are specified for cluster '{}', only one option can be used at the same time", key);
```

**Review by antonio2368 — APPROVED** [review]

**Comment by vdimir** [comment]
> Stateless tests (debug) [2/5] — fail: 2, passed: 1160, skipped: 10 [Details](https://s3.amazonaws.com/clickhouse-test-reports/58123/1f0a29cf10b43a641b3d1e284930491e7d7395c0/stateless_tests__debug__[2_5].html)

* https://github.com/ClickHouse/ClickHouse/issues/56931
* `01171_mv_select_insert_isolation_long`, `max_bytes_before_external_sort 1` in debug, test is slow

> Stress test (tsan) — Sanitizer assert (in stderr.log) [Details](https://s3.amazonaws.com/clickhouse-test-reports/58123/1f0a29cf10b43a641b3d1e284930491e7d7395c0/stress_test__tsan_.html)

https://github.com/ClickHouse/ClickHouse/issues/58071

**Pull request merged**